addons/website/static/tests/builder/translation.test.js

```diff
@@ -267,6 +272,23 @@ describe("save translation", () => {
     });
 });
 
+test("table of content snippet headings' translation updates its navbar items", async () => {
+    const snippet = "s_table_of_content";
+    const websiteContent = (await getStructureSnippet(snippet)).outerHTML;
+    const { getEditor } = await setupSidebarBuilderForTranslation({ websiteContent });
+    const editor = getEditor();
+    const oldTitle = editor.editable.querySelector("#table_of_content_heading_1_1").textContent;
+    expect(":iframe .s_table_of_content_navbar .table_of_content_link:first-child").toHaveText(
+        oldTitle
+    );
+    const titleEl = editor.editable.querySelector("#table_of_content_heading_1_1");
+    titleEl.textContent = "New title";
+    await manuallyDispatchProgrammaticEvent(titleEl, "input");
```

Comment:
You should use insertText (helper from html editor). This helper try to reproduce real user interaction